### PR TITLE
Issue 1095  callback improvements

### PR DESCRIPTION
**Pull request by adler-j**

### CODE REVIEW

File: odl/solvers/util/callback.py [hunk 1]
```diff
@@ -946,6 +946,65 @@ def __repr__(self):
...
+    def __repr__(self):
+        """Return ``repr(self)``."""
+        posargs = [self.niter]
+        optargs = [('step', self.step, 1)]
+        inner_str = signature_string(posargs, optargs)
+        if self.kwargs:
+            return '{}({}, **{})'.format(self.__class__.__name__,
+                                         inner_str)
+        else:
+            return '{}({})'.format(self.__class__.__name__,
```
Comment (kohr-h) [inline odl/solvers/util/callback.py]: Format string should be `{}({}, {})`

File: odl/solvers/util/callback.py [hunk 2]
```diff
@@ -946,6 +946,65 @@ def __repr__(self):
...
+        step : positive int, optional
+            Number of iterations between output. Default: 1
+        kwargs
+            Further parameters passed to ``tqdm.tqdm``.
+
+        Examples
+        --------
+        Create progress-bar with 10 iterations
+
+        >>> callback = CallbackProgressBar(niter=10)
```
Comment (kohr-h) [inline odl/solvers/util/callback.py]: Fails Travis who doesn't have `tqdm`. Not really sure how to circumvent this, though.

File: odl/solvers/util/callback.py [hunk 3]
```diff
@@ -946,6 +946,65 @@ def __repr__(self):
...
+
+    def __init__(self, niter, step=1, **kwargs):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        niter : positive int, optional
+            Total number of iterations.
+        step : positive int, optional
+            Number of iterations between output. Default: 1
```
Comment (kohr-h) [inline odl/solvers/util/callback.py]: Default clear from the signature.

File: odl/solvers/util/callback.py [hunk 4]
```diff
@@ -946,6 +946,65 @@ def __repr__(self):
...
+        kwargs
```
Comment (kohr-h) [inline odl/solvers/util/callback.py]: colon

File: odl/solvers/util/callback.py (hunk 2 above)
Comment (adler-j) [reply odl/solvers/util/callback.py]: I'll simply remove the doctests for now, use is frankly obvious

**Pull request merged**